### PR TITLE
Fixes in DownloadPersonalData

### LINKED ISSUE #14178: Identity -> DownloadPersonalData.cshtml.cs Remove dependencies on Newtonsoft.Json
now only dependencies Newtonsoft.Json in DownloadPersonalData.cshtml.cs.
maybe remove is better.

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/aspnet/AspNetCore/issues/14178 https://github.com/aspnet/AspNetCore/issues/5800

### CODE REVIEW

**Comment by Honjoy** [comment]
maybe change Encoding.UTF8.GetBytes(JsonSerializer.Serialize(personalData)) => JsonSerializer.SerializeToUtf8Bytes(personalData)  is better.

**Comment by HaoK** [comment]
Thanks for the suggestion @Honjoy

**Review by pranavkm — APPROVED** [review]

File: src/Identity/UI/src/Areas/Identity/Pages/V3/Account/Manage/DownloadPersonalData.cshtml.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public override async Task<IActionResult> OnPostAsync()
             personalData.Add($"Authenticator Key", await _userManager.GetAuthenticatorKeyAsync(user));
 
             Response.Headers.Add("Content-Disposition", "attachment; filename=PersonalData.json");
-            return new FileContentResult(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(personalData)), "text/json");
+            return new FileContentResult(JsonSerializer.SerializeToUtf8Bytes(personalData), "application/json");
```
Comment (pranavkm) [inline src/Identity/UI/src/Areas/Identity/Pages/V3/Account/Manage/DownloadPersonalData.cshtml.cs]: @HaoK does the project need to remove the package reference to Newtonsoft.Json?

**Review by pranavkm — CHANGES_REQUESTED** [review]
This one: https://github.com/aspnet/AspNetCore/blob/release/3.1/src/Identity/UI/src/Microsoft.AspNetCore.Identity.UI.csproj#L39

File: src/Identity/UI/src/Areas/Identity/Pages/V3/Account/Manage/DownloadPersonalData.cshtml.cs (hunk 1 above)
Comment (HaoK) [reply src/Identity/UI/src/Areas/Identity/Pages/V3/Account/Manage/DownloadPersonalData.cshtml.cs]: Yeah thanks good catch!

File: src/Identity/UI/ref/Microsoft.AspNetCore.Identity.UI.csproj [hunk 2]
```diff
@@ -5,7 +5,6 @@
   </PropertyGroup>
   <ItemGroup Condition="'$(TargetFramework)' == '$(DefaultNetCoreTargetFramework)'">
     <Compile Include="Microsoft.AspNetCore.Identity.UI.netcoreapp.cs" />
-    <Reference Include="Newtonsoft.Json"  />
```
[the remark this answers is not in the document]
Comment (pranavkm) [reply src/Identity/UI/ref/Microsoft.AspNetCore.Identity.UI.csproj]: Yeah, that's true. Should we just retarget this PR to 5.0? None of the changes here are useful if we cannot remove the package reference.

[the remark this answers is not in the document]
Comment (HaoK) [reply src/Identity/UI/ref/Microsoft.AspNetCore.Identity.UI.csproj]: Yeah might as well I'll retarget and move the bug to 5.0

**Comment by vanillajonathan** [comment]
Should it use `FileContentResult` or `JsonResult`?

**Pull request merged**